frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx

```diff
@@ -119,7 +119,7 @@ export const createKmsResources = (kms: KMSConfig, update = false, previousData?
     VAULT_CLIENT_CERT: kms.clientCert?.metadata.name,
     VAULT_CLIENT_KEY: kms.clientKey?.metadata.name,
     VAULT_NAMESPACE: kms.providerNamespace,
-    VAULT_TOKEN_NAME: KMSSecretName,
+    VAULT_TOKEN_NAME: createOcsKms ? KMSSecretName : '',
```

Comment:
If we split this into two functions we don't need these flags, and we are creating a lot of variables and we are not using those all the time.